### DISCUSSION TITLE
关于index.py的框架设计

### DISCUSSION THREAD

**Discussion by frostming** [abersheeran/index.py#28] [General]
有几个疑问和建议，想与你探讨。

我关注过最初版本的index.py，是类似于php的文件树形式的路由，后来还是变成了注册式的路由。我认为前一种设计是比较大的亮点，不知变更的考量是怎样。

其实最初的路由方法，还是不太理想，需要在文件里定义一个view类。在我的设想中，接近于Php体验的路由文件是这样的：

```python
# views/post.py

async def index(request):
    # 跟路由，对应/post/
    ...

# 可选装饰器，可以修改路由参数
# 默认url为函数名.replace('_', '-'), method=GET
@route(methods=["POST"], name="create")
async def create_post(request):
    # 对应url /post/create
    ...
```

只是一个建议，不知设计还有没可能再变，如果不能改了也没关系

**Comment by abersheeran** [comment]
路由的更改发生在 0.12，这个版本里我把底层路由从哈希表改成 Radix Tree，路由更加自由，快速，支持更多的路由注册方式，bottle 风格、django 风格、php 风格都可以做到。

https://index-py.abersheeran.com/master/route/#fileroutes 通过树状路由反向实现 FileRoutes，如此一来，FileRoutes 就能支持路径参数了。FileRoutes 是 Index.py 名称的由来之一😀除非我改名了，不然不会放弃的。

你这种定义也是可以做到的，参考 FileRoutes 的实现，新建一个 Routes 子类，在路由展开阶段读取文件内容，生成 Route 对象注册即可。

**Comment by frostming** [comment]
Cool，我看看
